### ISSUE TITLE
Uncaught (in promise) LinkError: import object field '__wbg_error_09919627ac0992f5' is not a Function

### ISSUE THREAD

**Issue by Niedzwiedzw** [rustwasm/wasm-bindgen#2779]
that happens when I import a package generated by wasm-pack

**Comment by Niedzwiedzw** [comment]
https://github.com/nshen/vite-plugin-wasm-pack this partially solved it

**Comment by Niedzwiedzw** [comment]
now the problem is that despite me passing an object into a function, what goes into wasm is read as `null`. 
invalid type: null, expected struct ScheduleImportConfigV2 at line 1 column 4

**Comment by Niedzwiedzw** [comment]
yeah so I've triple checked it with `console.log(myInput)` on javascript side just beforre passing that into a wasm function
```ts
console.log(myInput) // {a: 1, b: 2}
myWasmPackage.handle_my_input(myInput)
```

and in rust it's 
```rust
#[wasm_bindgen(skip_typescript)]
pub fn handle_my_input(my_data: JsValue) -> JsResult<JsValue> {
    console_log!(
        "handle_my_input :: {:#?}",
        my_data
    );
    // ...

}

```
and it prints 
```
handle_my_input :: JsValue(undefined)
```

**Comment by Niedzwiedzw** [comment]
https://github.com/Niedzwiedzw/vite-wasm-pack-error-showcase here's a repository showcasing this problem

**Comment by Menci** [comment]
I'm attempting to reproduce it but encountered an error building your Rust code:

<img width="1143" alt="image" src="https://user-images.githubusercontent.com/11341955/156889430-b4c3790d-72cc-4083-839c-e1f5bccea714.png">

Could you please test it using my plugin https://www.npmjs.com/package/vite-plugin-wasm and the `bundler` target?

**Comment by Jieiku** [comment]
I ran into this Q4 2023 but was unable to figure out the cause, and because of that I quit using tinysearch.

I opened the issue here: https://github.com/tinysearch/tinysearch/issues/175

I setup a reproducible test case repo for this here: https://github.com/Jieiku/abridge-tinysearch-dev

The readme in that test case repo has instructions for reproducing this error, I am stumped as to what to do next because I have almost zero rust experience.
